### ISSUE TITLE
An error occurred: NestedStack - Embedded stack xyz was not successfully created: The following resource(s) failed to create: [...]

### ISSUE THREAD

**Issue by GunterJameda** [dougmoscrop/serverless-plugin-split-stacks#84]
Hi!

I was trying to develop my API and I got an error like this one:

```
An error occurred: APINestedStack - Embedded stack arn:aws:cloudformation:eu-west-3:628286138150:stack/albumteller-api-dev-APINestedStack-KZFRR07OOUFR/7aa793f0-d94e-11e9-a439-064a2a94b120 was not successfully created: The following resource(s) failed to create: [ApiGatewayResourceLicensingDeviceTokenVar, ApiGatewayResourceLicensingValidate, ApiGatewayResourceProofingProofidVarContent, ApiGatewayResourceProofingProofidVarUpdate, ApiGatewayResourceLicensingCustomer, ApiGatewayResourceProofingProofidVarLoginDashtoken, ApiGatewayResourceProofingProofidVarTimeline]
```

I tried to delete the stack several times in AWS, I tried to change the configs for the plugin to see if could be something related to the number of nested stacks or something, but nothing worked.

I read, some time ago, that the best choice would always be to split the API in several ones smaller, and after reading #52, I got confirmation on what I read. The thing is: I didn't want to do that, as it is a last resource measure.

What am I possibly doing wrong?

Thanks!

PS: If you need any more info to understand whats going on, I'll gladly give it. :)

**Comment by jeremyputeaux** [comment]
Hello,
I'm having the same issue. Did you find a solution?
Thanks!

**Comment by GunterJameda** [comment]
Actually, I didn't.

I separated my apis in several projects so we don't have this kind of problem again. It might not be a very good idea for you, but it surely worked for me like a charm.

**Comment by jeremyputeaux** [comment]
Actually, I've found a solution to my problem.
My graphQL schema was incorrect.
I copy pasted it in the AWS AppSync to check its validity and there was 1 problem. I fixed it and it worked.
I hope it can help someone.

**Comment by GunterJameda** [comment]
I don't think it would be the same problem to me, anyway. I'm using mongoose as a ORM and my schemas did not change when the error start to show up, so, it might have been something else.

**Comment by dougmoscrop** [comment]
Sorry for the delay. You have to watch the Events tab of the nested stack as the deploy goes on to see more logs about why the resources are failing - you can't wait until after the deploy falls because often the stack disappears.  Watch it live.

**Comment by GunterJameda** [comment]
@dougmoscrop thanks for your reply.

When I was debugging the cause of failure, I did that often, because AWS eliminates the stacks in case of error as you said. Although I was watching, I couldn't see anything weird. It would just fail in some random stack and then start to rollback everything until there was another error on deletion. It was very very weird. I can't remember everything that went by, as I already moved on to another solution, as I previously told to @jeremyputeaux, but it was pretty much what I just described.

Are you sure there's nothing wrong with the plugin? I'm not saying there is or there isn't. It's just to prevent any error that might be there and we are not aware about.

**Comment by GunterJameda** [comment]
Well, it will be kinda hard now, as my project is way different than it was a month ago due to refactor. I would gladly provide that info, but it will be impossible, now. I'm sorry. :/

**Comment by suryacaprice** [comment]
is there any solution to this?

**Comment by GunterJameda** [comment]
There's a workaround:

https://github.com/dougmoscrop/serverless-plugin-split-stacks/issues/84#issuecomment-547824082

**Comment by lucasklaassen** [comment]
For debugging this, it's super helpful to pop into cloudformation and filter your stacks by Deleted. You'll be able to look at the event logs for the stack that was attempted to be created and was throwing errors:

<img width="1510" alt="Screen Shot 2020-03-26 at 9 47 54 AM" src="https://user-images.githubusercontent.com/6452107/77673194-18f08b00-6f47-11ea-8eeb-a0d28039b76f.png">

**Comment by NairiAreg** [comment]
So, did anybody fix this? 
I have the same error

**Comment by mathmarqq** [comment]
I had the same issue. You should disable the rollback when you deploy, then check the errors on Cloudformation, the main error is something like "X was not successfully created: the following resources.... A, B and C failed to create", so you should check the resources A, B and C and see why the creation failed.

In my case, I was deploying by resource type, and I was reaching the quota limit of Cloudwatch Logs, so I changed the split type
